datastore/src/main/java/com/ds/datastore/Utilities.java

```diff
@@ -2,61 +2,75 @@
+import io.github.resilience4j.circuitbreaker.annotation.CircuitBreaker;
+import io.github.resilience4j.retry.annotation.Retry;
+import org.springframework.stereotype.Component;
 
 import java.net.URI;
 import java.net.http.HttpClient;
 import java.net.http.HttpRequest;
 import java.net.http.HttpResponse;
+import java.net.http.HttpRequest.Builder;
+import java.time.Duration;
+import java.util.Optional;
 
-public class Utilities {
 
-    public static HttpResponse<String> createGetConnection(String address, String serverAddress, Long id) throws Exception {
-        HttpRequest request = HttpRequest.newBuilder()
-                .uri(new URI(address))
-                .headers("Content-Type", "application/json;charset=UTF-8")
-                .setHeader("referer", serverAddress)
-                .setHeader("id", String.valueOf(id))
-                .GET()
-                .build();
-        return HttpClient.newBuilder()
-                .build()
-                .send(request, HttpResponse.BodyHandlers.ofString());
-    }
+@Component
+public class Utilities {
 
-    public static HttpResponse<String> createPostConnection(String address, JsonObject jso) throws Exception {
+    @Retry(name = "retry")
+    public HttpResponse<String> createConnection(String address, JsonObject jso, String serverAddress, Long id, String requestType) throws Exception{
+        System.out.println("utilities called");
         Gson gson = new Gson();
         String json = gson.toJson(jso);
-        HttpRequest request = HttpRequest.newBuilder()
-                .uri(new URI(address))
-                .headers("Content-Type", "application/json;charset=UTF-8")
-                .POST(HttpRequest.BodyPublishers.ofString(json))
-                .build();
-        return HttpClient.newBuilder()
+        Builder builder = HttpRequest.newBuilder()
+            .uri(new URI(address))
+            .headers("Content-Type", "application/json;charset=UTF-8");
+        if (requestType.equals("GET")) {
+                builder = builder
+                        .setHeader("referer", serverAddress)
+                        .setHeader("id", String.valueOf(id))
+                        .timeout(Duration.ofSeconds(4))
+                        .GET();
+        }else if(requestType.equals("POST")){
+                builder = builder.POST(HttpRequest.BodyPublishers.ofString(json));
+        }else if(requestType.equals("PUT")){
+                builder = builder.PUT(HttpRequest.BodyPublishers.ofString(json));
+        }else if (requestType.equals("DELETE")) {
+                builder = builder.DELETE();
+        }
+        HttpRequest request = builder.build();
+        HttpResponse<String> response = HttpClient.newBuilder()
                 .build()
                 .send(request, HttpResponse.BodyHandlers.ofString());
+        if(requestType.equals("POST") && (response.statusCode() != 201 && response.statusCode() != 200) ){
+                System.out.println(response.statusCode());
```

Comment:
lets log something more helpful here and return a more helpful exception (with some content)